audit/audit-gcp.sh

```diff
@@ -0,0 +1,92 @@
+
+format=json
+gcloud iam roles list --organization=$CNCF_GCP_ORG --format=$format \
+       > cncf-org.roles.$format
+gcloud organizations get-iam-policy $CNCF_GCP_ORG --format=$format \
+       > cncf-org.policy.$format
+gcloud projects list \
+       --filter "parent.id=$CNCF_GCP_ORG" \
+       --format "value(name, projectNumber)" \
+    | while read PROJECT NUM; do \
+    export CLOUDSDK_CORE_PROJECT=$PROJECT
+    gcloud projects get-iam-policy $PROJECT --format=$format > $PROJECT.policy.$format
+    gcloud iam roles list --project $PROJECT --format=$format > $PROJECT.roles.$format
+    mkdir -p $PROJECT.roles
+    for ROLE_PATH in `gcloud iam roles list --project $PROJECT --format="value(name)"`
+    do
+        ROLE=`basename $ROLE_PATH`
+        gcloud iam roles --project=$PROJECT describe $ROLE \
+               --format=json > $PROJECT.roles/$ROLE.json
+    done
+    gcloud services list --filter state:ENABLED --format=$format > $PROJECT.services.$format
+    for service in `gcloud services list --filter state:ENABLED --format=json | jq -r .[].config.name`
+    do
+        case $service in
+            compute.googleapis.com)
+                echo TODO: Needs compute.projects.get
+                #### gcloud compute project-info describe
+                #### gcloud compute instances list --format=$format > $PROJECT.compute.instances.$format
+                #### gcloud compute disks list --format=$format > $PROJECT.compute.disks.$format
+                # I'm ensure why we see this when container.googleapis.com is DISABLED
+                gcloud container clusters list --format=$format > $PROJECT.clusters.$format
+                ;;
+            dns.googleapis.com)
+                mkdir -p dns
+                gcloud dns project-info describe $PROJECT --format=$format > dns/$PROJECT.info.$format
+                gcloud dns managed-zones list --format=$format > dns/$PROJECT.zones.$format
+                ;;
+            logging.googleapis.com)
+                echo TODO: Needs serviceusage.services.use
+                ##### gcloud logging logs list --format=$format > $PROJECT.logging.logs.$format
+                ##### gcloud logging metrics list
+                ##### gcloud logging sinks list
+                ;;
+            monitoring.googleapis.com)
+                echo TODO: Needs serviceusage.services.use
+                #### gcloud alpha monitoring policies list
+                #### gcloud alpha monitoring channels list
+                #### gcloud alpha monitoring channel-descriptors list
+                ;;
+            oslogin.googleapis.com)
+                echo TODO: Verify how OS Login is configured / audited
+                ;;
+            bigquery-json.googleapis.com)
+                echo TODO: Verify how Big Query is configured / audited
+                ;;
+            storage-api.googleapis.com)
+                echo TODO: Add storage.buckets.get for auditors
```

Comment:
Need storage.buckets.get to ensure bucket-policy-only

Reply:
Added